pkg/controllermanager/controller/shoot/shoot_hibernation_control_test.go

```diff
@@ -259,6 +278,28 @@ var _ = Describe("Shoot Hibernation", func() {
 							Enabled: &enabled,
 						}))
 					}),
+					recorder.EXPECT().Eventf(&shoot, corev1.EventTypeNormal, gardencorev1beta1.ShootEventHibernationEnabled, "%s", "Hibernating cluster due to schedule").Times(1),
+				)
+
+				job.Run()
+			})
+
+			It("should set the hibernation status correctly to disabled", func() {
+				enabled := false
+				job := NewHibernationJob(clientMap, log, recorder, &shoot, enabled)
+
+				gomock.InOrder(c.EXPECT().CoreV1beta1().Return(gardenIface),
+					gardenIface.EXPECT().Shoots(namespace).Return(shootIface),
+					shootIface.EXPECT().Get(name, metav1.GetOptions{}).Return(&shoot, nil),
+
+					c.EXPECT().CoreV1beta1().Return(gardenIface),
+					gardenIface.EXPECT().Shoots(namespace).Return(shootIface),
+					shootIface.EXPECT().Update(gomock.AssignableToTypeOf(&gardencorev1beta1.Shoot{})).Do(func(actual *gardencorev1beta1.Shoot) {
+						Expect(actual.Spec.Hibernation).To(Equal(&gardencorev1beta1.Hibernation{
+							Enabled: &enabled,
+						}))
+					}),
+					recorder.EXPECT().Eventf(&shoot, corev1.EventTypeNormal, gardencorev1beta1.ShootEventHibernationDisabled, "%s", "Waking up cluster due to schedule").Times(1),
```

Comment:
Just FYI, you don't have to specify `.Times(1)`. If no count is specified, it's assumed to be `1`.
Not saying, that you should change it, it's ok to be explicit. Just to let you know :)